### PR TITLE
(Openstack) Security Group UI Code.

### PR DESCRIPTION
**Pull request by gadresushrut**
My pull request is dependent on the pull request created by jcwest [PR No. 2384] 

Creating openstack security groups  is done.

Details and Edit functionality are not implemented yet, as backend code is not ready.

### CODE REVIEW

File: app/scripts/modules/openstack/securityGroup/configure/wizard/rules.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        </tr>
+        </thead>
+        <div>
+          <tbody>
+            <tr ng-repeat="rule in securityGroup.rules">
+              <td><select class="form-control input-sm" ng-model="rule.type" ng-init="rule.type='tcp'"
+                          ng-options="protocol as protocol.toUpperCase() for protocol in ['tcp']" required></select></td>
+              <td><input class="form-control input-sm" type="number" ng-init="rule.fromPort=80" min="0" ng-model="rule.fromPort" required/></td>
+              <td><input class="form-control input-sm" type="number" ng-init="rule.toPort=80" min="0" ng-model="rule.toPort" required/></td>
+              <td><input class="form-control input-sm" type="text" ng-init="rule.cidr='0.0.0.0/0'" ng-model="rule.cidr" required/></td>
```
Comment (anotherchrisberry) [inline app/scripts/modules/openstack/securityGroup/configure/wizard/rules.html]: Pretty sure all the `ng-init`s are going to override the `ng-model` value. If you want these to have a default value, you should put those defaults into `openstackSecurityGroupTransformer#constructNewIngressRule`.

File: app/scripts/modules/openstack/securityGroup/configure/wizard/rules.html [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          <th></th>
+        </tr>
+        </thead>
+        <div>
+          <tbody>
+            <tr ng-repeat="rule in securityGroup.rules">
+              <td><select class="form-control input-sm" ng-model="rule.type" ng-init="rule.type='tcp'"
+                          ng-options="protocol as protocol.toUpperCase() for protocol in ['tcp']" required></select></td>
+              <td><input class="form-control input-sm" type="number" ng-init="rule.fromPort=80" min="0" ng-model="rule.fromPort" required/></td>
+              <td><input class="form-control input-sm" type="number" ng-init="rule.toPort=80" min="0" ng-model="rule.toPort" required/></td>
```
Comment (anotherchrisberry) [inline app/scripts/modules/openstack/securityGroup/configure/wizard/rules.html]: If you change `min` from `"0"` to `"{{rule.fromPort}}"`, it should take care of the TODO below.

(You may want to add a validation message, but hopefully it'll be obvious to the user that the value is invalid)

Comment (gadresushrut) [inline app/scripts/modules/openstack/securityGroup/configure/wizard/rules.html]: I did the same thing first, but I was not able to display validation message, so though to take same different approach.
Hence added TODO.

I will again do the change to replace 0 with rule.fromPort and try to display validation message. If validation message is not possible can we omit that message.

**Comment by gadresushrut** [comment]
I have addressed all the comments mentioned by you.

File: app/scripts/modules/openstack/securityGroup/configure/wizard/rules.html [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+          <th style="width: 15%">From Port</th>
+          <th style="width: 15%">To Port</th>
+          <th style="width: 15%">CIDR</th>
+          <th></th>
+        </tr>
+        </thead>
+        <div>
+          <tbody>
+            <tr ng-repeat="rule in securityGroup.rules">
+              <td><select class="form-control input-sm" ng-model="rule.type" ng-init="rule.type='tcp'"
```
Comment (anotherchrisberry) [inline app/scripts/modules/openstack/securityGroup/configure/wizard/rules.html]: Should not use `ng-init` here.

**Comment by anotherchrisberry** [comment]
Needs to be rebased into a single commit.

**Pull request closed without merging**